plugins/inputs/directory_monitor/README.md

```diff
@@ -0,0 +1,59 @@
+# Directory Monitor Input Plugin
+
+This plugin monitors a single directory (without looking at sub-directories), and takes in each file placed in the directory.
+The plugin will gather all files in the directory at a configurable interval (`monitor_interval`), and parse the ones that haven't been picked up yet.
+
+Please be advised that this plugin pulls files directly after they've been in the directory for the length of the configurable `directory_duration_threshold`, and thus files should not be written 'live' to the monitored directory unless they are guaranteed to finish writing before the `directory_duration_threshold`. This plugin is intended to read files that are moved or copied to the monitored directory, and thus files should also not be used by another process or else they may fail to be gathered.
+
+### Configuration:
+
+```toml
+## The directory to monitor and read files from.
+directory = ""
+#
+## The directory to move finished files to.
+finished_directory = "5s"
+#
+## Whether or not to move files that error out to an error directory.
+use_error_directory = "true"
+#
+## The directory to move files to upon file error, given that 'use_error_directory' is enabled.
+## If not is given, the error directory will be auto-generated.
+# error_directory = ""
+#
+## The interval at which to check the directory for new files.
+# monitor_interval = "50ms"
+#
+## The amount of time a file is allowed to sit in the directory before it is picked up.
+## This time can generally be low but if you choose to have a very large file written to the directory and it's potentially slow,
+## set this higher so that the plugin will wait until the file is fully copied to the directory.
+# directory_duration_threshold = "50ms"
+#
+## Character encoding to use when interpreting the file contents. Invalid
+## characters are replaced using the unicode replacement character. Defaults to utf-8.
+##   ex: character_encoding = "utf-8"
+##       character_encoding = "utf-16le"
+##       character_encoding = "utf-16be"
+# character_encoding = "utf-8"
```

Comment:
Line protocol can't process anything other than UTF-8, so if we want to support others we will need to convert them to UTF-8 anyway. Not sure we want to make that the responsibility of this plugin.